File: Data Structures & Algorithms/daily-temperatures/submission-0.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        vector<int> ans(temperatures.size(), 0);
        stack<int> x;
        for(int i{0}; i < temperatures.size(); ++i){
            while(!(x.empty()) && temperatures[i] > temperatures[x.top()]){
                int temp{x.top()};
                x.pop();
                ans[temp] = i-temp;
            }
            x.push(i);
        }
        return ans;
    }
};

File: Data Structures & Algorithms/transpose-matrix/submission-0.java
class Solution {
    public int[][] transpose(int[][] matrix) {
        if(matrix.length == matrix[0].length){
            for(int i = 0; i < matrix.length; i++){
                for(int j = i+1; j < matrix[0].length; j++){
                    int temp = matrix[i][j];
                    matrix[i][j] = matrix[j][i];
                    matrix[j][i] = temp;
                }
            }
            return matrix;
        }
        int[][] ans = new int[matrix[0].length][matrix.length];
        for(int i = 0; i < matrix.length; i++){
            for(int j = 0; j < matrix[0].length; j++){
                ans[j][i] = matrix[i][j];
            }
        }
        return ans;
    }
}